src/chocolatey/infrastructure/information/ProcessInformation.cs

```diff
@@ -158,6 +161,52 @@ public static bool UserIsSystem()
             return isSystem;
         }
 
+        public static ProcessTree GetProcessTree(Process process = null)
+        {
+            if (process == null)
+            {
+                process = Process.GetCurrentProcess();
+            }
+
+            var tree = new ProcessTree(process.ProcessName);
+
+            Process nextProcess = null;
+
+            while (true)
+            {
+                try
+                {
+                    var parentProcess = ParentProcessUtilities.GetParentProcess(nextProcess ?? process);
+
+                    if (parentProcess == null)
+                    {
+                        break;
+                    }
+
+                    nextProcess = parentProcess;
+                    tree.Processes.AddLast(nextProcess.ProcessName);
+                }
+                catch (Win32Exception ex)
+                {
+                    // Native error code 5 is access denied.
+                    // This usually happens if the parent executable
+                    // is running as a different user, in which case
+                    // we are not able to get the necessary handle for
+                    // the process.
+                    if (ex.NativeErrorCode != 5)
+                    {
+                        throw;
+                    }

```

Comment:
Alternately maybe this should just write a different error message, rather than rethrowing and making it necessary for us to handle elsewhere?